modules/planning/navi/common/local_path.cc

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+ * Copyright 2018 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+/**
+ * @file
+ * @brief This file provides the implementation of the class "LocalPath".
+ */
+#include <memory>
+#include <vector>
+
+#include "modules/planning/navi/common/local_path.h"
+
+namespace apollo {
+namespace planning {
+
+using apollo::common::math::Vec2d;
+
+std::shared_ptr<double> LocalPath::GetInitY() {
+  if (points_.empty()) {
+    return nullptr;
+  }
+  std::shared_ptr<double> y_ptr;
+  *y_ptr = points_[0].y();
+  return y_ptr
+}
+
+std::vector<Vec2d> LocalPath::GetXYPoints() { return points_; }
+
+int LocalPath::GetRange() { return static_cast<int>(path_points_.size() - 1); }
+
+void LocalPath::Shift(const double dist) {
+  for (int i = 0; i < path_points_.size(); ++i) {
+    path_points_[i].set_y(path_points_[i].y() + dist);
+  }
+}
+
+void LocalPath::Cut(const double dist) {
+  // TODO(all):not realized now and will be perfected in the future
+  return;
+}
+
+void LocalPath::Resample() {
+  // TODO(all):not realized now and will be perfected in the future
+  return;
+}
+
+void LocalPath::Merge(const common::Path local_path, const double weight) {
+  for (size_t i = 0; i < path_points_.size(); ++i) {
+    auto y = path_points_[i].y();
+    if (i < local_path.path_point_size()) {
```

Comment:
static_cast<int>(i) < local_path.path_point_size().
In protobuf, the size is defined as an int type. We should avoid comparing between different types.